openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -55,18 +58,16 @@ def __init__(self, course_key, request, username=''):
             username or request.user.username,
             course_key,
         )
-        self.effective_user = self.overview.effective_user
```

Comment:
Some big changes in this __init__ method. I posted in TNL to also take a look at these and make sure it doesn't break their flow.